WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingViewModel.swift

```diff
@@ -44,47 +51,62 @@ class StoreOnboardingViewModel: ObservableObject {
 
     let isExpanded: Bool
 
-    private let siteID: Int64
+    private var siteID: Int64?
 
     private let stores: StoresManager
 
     private var state: State
 
-    private let placeholderTasks: [StoreOnboardingTaskViewModel] = Array(repeating: StoreOnboardingTaskViewModel.placeHolder(),
-                                                                         count: 3)
+    private let placeholderTasks: [StoreOnboardingTaskViewModel] = [.placeHolder(), .placeHolder(), .placeHolder()]
 
     private let defaults: UserDefaults
 
+    /// Emits when there are no tasks available for display after reload.
+    /// i.e. When (request failed && No previously loaded local data available)
+    ///
+    @Published private var noTasksAvailableForDisplay: Bool = false
+
     /// - Parameters:
     ///   - isExpanded: Whether the onboarding view is in the expanded state. The expanded state is shown when the view is in fullscreen.
-    ///   - siteID: siteID
     ///   - stores: StoresManager
-    ///   - userDefaults: UserDefaults for storing when all onboarding tasks are completed
+    ///   - defaults: UserDefaults for storing when all onboarding tasks are completed
     init(isExpanded: Bool,
-         siteID: Int64,
          stores: StoresManager = ServiceLocator.stores,
          defaults: UserDefaults = .standard) {
         self.isExpanded = isExpanded
-        self.siteID = siteID
         self.stores = stores
         self.state = .loading
         self.defaults = defaults
+
+        Publishers.CombineLatest($noTasksAvailableForDisplay,
+                                 defaults.publisher(for: \.completedAllStoreOnboardingTasks))
+        .map { !($0 || $1) }
+        .assign(to: &$shouldShowInDashboard)
     }
 
-    func reloadTasks() async {
+    func reloadTasks(siteID: Int64) async {
```

Comment:
nit: I don't think this is necessary, since the siteID stays constant as in `DashboardViewController.siteID`. Only the `Site` object (other properties about the site) can change. The siteID can be DI'ed in the initializer like before.

Reply:
Oh, good point. Done in 9b94f291a1e2dc94713d3ee5acf12d1dec9bb34e